araproc/analysis/standard_reco.py

```diff
@@ -582,34 +583,22 @@ def get_surface_corr_max(self, corr_map, z_thresh=-10):
         if hist is None:
             raise ValueError("The 'map' key was not found in corr_map.")
         
-        surf_corr_values = []
-        corr_bins = []
-
-        # Loop through the bins and store values within theta range
-        for x_bin in range(1, hist.GetNbinsX() + 1):
-            for y_bin in range(1, hist.GetNbinsY() + 1):
-                theta = hist.GetYaxis().GetBinCenter(y_bin)
-                phi = hist.GetXaxis().GetBinCenter(x_bin)
-                corr = hist.GetBinContent(x_bin, y_bin)
-                if theta_min <= theta <= theta_max:
-                    surf_corr_values.append(corr)
-                    corr_bins.append((x_bin, y_bin))
-
-        # Identify maximum surface correlation
-        if not surf_corr_values:
-            raise RuntimeError("No correlation values found in the specified theta range.")
+        # set the range of the histogram to only be where it's relevant
+        theta_min_bin = hist.GetYaxis().FindBin(theta_min)
+        theta_max_bin = hist.GetYaxis().FindBin(theta_max)
+        hist.GetYaxis().SetRange(theta_min_bin+1, theta_max_bin-1)
         
-        # Check for NaNs
-        if np.isnan(surf_corr_values).any():
```

Comment:
I actually have been getting this error sometimes. Do you know why? Is it safe to remove this?

Reply:
I have absolutely no idea why this would happen...

Reply:
Ya this is weird, it looks like it would only happen if the map itself had nans in it. I guess that's not the RTC behavior?

Reply:
That's right. The map should never have nan's in it...

Reply:
If it does, I'd like to know so I can address it. Can you identify a specific event where this is firing?

Reply:
It fires on some bad quality events, but it also fired on event index 5 of run 5406 on A5 which is apparently a good quality event.

Reply:
This is the path if you want to check it out `/data/exp/ARA/2019/blinded/L1/ARA05/0605/run005406/event005406.root`

Reply:
Nevermind, I can't reproduce this. Maybe AraRoot was being recompiled or something and it got borked.